lineax/_operator.py

```diff
@@ -1514,7 +1514,11 @@ def is_diagonal(operator: AbstractLinearOperator) -> bool:
 @is_diagonal.register(JacobianLinearOperator)
 @is_diagonal.register(FunctionLinearOperator)
 def _(operator):
-    return diagonal_tag in operator.tags
+    return (
+        diagonal_tag in operator.tags
+        or operator.in_size() == 1
+        and operator.out_size() == 1
```

Comment:
I think it'd be good practice to group these in brackets: `diagonal_tag in operator.tags or (operator.in_size() == 1 and operator.out_size() == 1)`

I don't have the precedence rules for `x or y and z` in my head!

Reply:
Done!